staging/src/k8s.io/apiserver/pkg/admission/plugin/authorizer/caching_authorizer.go

```diff
@@ -39,7 +39,10 @@ type cachingAuthorizer struct {
 	decisions  map[string]authzResult
 }
 
-func newCachingAuthorizer(in authorizer.Authorizer) authorizer.Authorizer {
+// NewCachingAuthorizer returns and author that caches decisions for the duration
+// of the authorizers use.  Intended to be used for short-lived operations such as
+// the handling of a request in the admission chain, and then discarded.
```

Comment:
am curious, what is the reason we need this on-top of a "normal" caching authorizer?

Reply:
This is the authorizer added by https://github.com/kubernetes/kubernetes/pull/116443.  I'm not familiar with how it compares with the "normal" one, but it's targeted specifically at caching only for the scope of a single request's admission chain handling, and is intended to cache results for identical admission checks performed from CEL.